src/plugins/controls/public/time_slider/embeddable/time_slider_embeddable.tsx

```diff
@@ -141,8 +141,8 @@ export class TimeSliderControlEmbeddable extends Embeddable<
           timeRangeBounds: nextBounds,
         })
       );
-      this.syncWithTimeRange();
     }
+    this.syncWithTimeRange();
```

Comment:
The timeslider control wasn't actually discarding the changes to the selected time range value on input change because it was **only** syncing with the time range when the global time range changed. Moving this `syncwithTimeRange` call outside of the `if` statement fixed this :+1:

Reply:
this.syncWithTimeRange() should stay in the if statement so syncWithTimeRange is not called every time input changes.

I could not reproduce the problem you mentioned and was able to do the following
1) edit code, moving syncwithTimeRange back into if statement
2) create dashboard with timeslider control
3) save dashboard
4) switch to view mode
5) switch back to edit mode
6) change time range - verified time slider control updated to new time range
7) Click "Switch to view mode" and select discard changes in modal
8) verified time slider control updated to previous time range.

Reply:
Oh... hrm. Very odd.

Here is me replicating the behaviour when I move the `syncwithTimeRange` back into the `if`:


https://user-images.githubusercontent.com/8698078/207974872-37e748e6-ce7b-45b4-8018-1411e4a4888a.mov

I will try with a freshly bootstrapped Kibana to see if something funky is happening....

Reply:
Yup, tried it with a fresh install and I can still replicate that the changes to the time slider are **not** being discarded. @nreese I'm not sure why you aren't able to replicate it? Very odd....

Reply:
Thank you for the video, that helped. I was changing the time range in the global time picker, not changing the selected timeslice in the time slider. I view the problem now.

Calling syncWithTimeRange on any input change is a problem. syncWithTimeRange publishes timeslice and updates a bunch of redux state that does not need to be updated.

How about wrapping syncWithTimeRange in an if statement to only call when this case happens. See https://github.com/elastic/kibana/commit/97887a564fb9c7207f55820a644c1e8ad9e7ebc8 as an example.

Reply:
Good idea! :) I added something along those lines in 62f2f0d61d1e17af7083020c34744296f78f663f

It's definitely a bit messy - not sure if there is a better way to go about it. Because `componentState` stores the value but `explicitInput` stores the percentage, there isn't a super easy way to reset the component state back to its previous value when the `explicitInput` is changed - that is why I'm still calling `syncWithTimeRange` rather than setting the component state directly in this case. Unless we want to split out all the `to` and `from` calculations for setting `value` to be separate from `syncWithTimeRange`, then always call this new `updateComponentStateValue` as part of `onInputChanged`.... Interested in your thoughts!